src/pages/VideoDetailsView/VideoDetailsView.page.jsx

```diff
@@ -0,0 +1,97 @@
+import { useYouTube } from '../../utils/hooks/useYouTube';
+import { useStoreContext } from '../../state/Store.provider';
+import { storage } from '../../utils/storage';
+import { FAVORITES_LIST_STORAGE_KEY } from '../../utils/constants';
+import {
+  FavoritesBtn,
+  Title,
+  Description,
+  Video,
+  VideoDetails,
+  Grid,
+} from './VideoDetailsView.styles';
+
+function VideoDetailsView() {
+  const [buttonAction, setButtonAction] = useState('Add');
+  const { id } = useParams();
+  const location = useLocation();
+  const { videoTitle, videoDescription, image } = location.state;
+  const url = 'https://www.youtube.com/embed/' + id;
+  const { store } = useStoreContext();
+  const { searchTerm } = store;
+  const FavoritesList = storage.get(FAVORITES_LIST_STORAGE_KEY);
+  const favoriteVideo = {
+    id: {
+      kind: 'youtube#video',
+      videoId: id,
+    },
+    snippet: {
+      title: videoTitle,
+      description: videoDescription,
+      thumbnails: {
+        high: {
+          url: image,
+        },
+      },
+    },
+  };
+  const firstFavorite = {
+    items: [favoriteVideo],
+  };
+
+  useEffect(() => {
+    if (FavoritesList?.items) {
+      //Add or Remove?
+      let filterFavorites = FavoritesList.items.filter((e) => e.id.videoId === id);
+      if (filterFavorites.length === 1) {
+        setButtonAction('Remove');
+      } else {
+        setButtonAction('Add');
+      }
+    }
+  }, [FavoritesList, id]);
+  function handleClick() {
+    if (buttonAction === 'Add') {
+      if (!FavoritesList?.items) {
+        //First Use
+        storage.set(FAVORITES_LIST_STORAGE_KEY, firstFavorite);
+      } else {
+        FavoritesList.items[FavoritesList.items.length] = favoriteVideo;
+        storage.set(FAVORITES_LIST_STORAGE_KEY, FavoritesList);
+      }
+      setButtonAction('Remove');
+    } else if (buttonAction === 'Remove') {
+      var index = FavoritesList.items.findIndex(function (e, i) {
+        return e.id.videoId === id;
+      });
+      FavoritesList.items[index].id.videoId = 'deleted';
+      FavoritesList.items[index].id.kind = 'youtube#channel';
+      storage.set(FAVORITES_LIST_STORAGE_KEY, FavoritesList);
+      setButtonAction('Add');
+    }
+  }
+  return (
+    <Grid data-testid="location-videoDetailsView" className="grid">
+      <VideoDetails>
+        <Video src={url} frameBorder="0" allowFullScreen></Video>
+        <Title>{videoTitle}</Title>
+        {store.sessionData.id === '' ? (
+          <></>
+        ) : (
+          <>
+            <FavoritesBtn data-testid="favoritesBtn" onClick={handleClick}>
+              {buttonAction === 'Add' ? 'Add to favorites' : 'Remove from favorites'}
+            </FavoritesBtn>
+          </>
+        )}
+        <Description>{videoDescription}</Description>
+      </VideoDetails>
+      <ListOfRelatedVideos videos={useYouTube(searchTerm)}></ListOfRelatedVideos>
+    </Grid>
+  );
+}
+
+export default VideoDetailsView;
```

Comment:
You have some code that could be reused in `FavoriteVideoDetailsView` and `VideoDetailsView`.  I recommend instead of two total components you create an extra one that has everything that is shared between these two components and that everything that is different is sent down as props from the two existing components.

Your new component should be the one that is roughly 100 lines long and the existing two should be way shorter as they will only do the things that are particular for them and everything shared is in the new component.

This might benefit from a file renaming as well:

VideoDetailsView -> MainVideoDetailsView
FavoriteVideoDetailsView -> stays the same
new component -> VideoDetailsView